### ISSUE TITLE
Data Views Grid layout: The focusable preview has no visible focus style

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#62958]
### Description

In the grid layout, items are presented with a preview (or with a grey empty area) which is also a button.
If the item is editable, activating the button opens the editor.
If the item is not editable, the button is disabled.

When you hover over an editable item, the cursor changes to the hand, but it is the only visual indicator that the item preview is hovered.
When you use the keyboard to navigate between focusable elements on the page, there is no visual indicator that the item preview is focused.

This means that is is not possible to see where the current focus is.
For example, if you move focus to the "View options" option and press the tab key, the focus is on the first item preview, but it is not visible, and pressing tab again moves focus to the checkbox next to the item name.

### Step-by-step reproduction instructions

Activate a block theme.
Open Editor > Termplates.
Make sure the Grid layout is active.
On this screen, place the cursor in the search field.
Use the tab key repeatedly to move forwards.
Notice that when you reach the list of templates, the preview has no visible focus style.

Repeat the test for pages and patterns.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
I was able to reproduce this issue as well. In WP6.5, the focus style displays correctly, as shown below.

![image](https://github.com/WordPress/gutenberg/assets/54422211/7bbeab4a-71de-4df5-bfe5-3b3d27e3a974)

I believe this is a regression that first appeared in WP6.6, so I'll add it to the project board.
